core/src/test/java/org/infinispan/security/DynamicRBACTest.java

```diff
@@ -107,6 +114,33 @@ public void testClusterPermissionMapper() {
       assertEquals(1, roles.size());
    }
 
+   public void testJoiner() throws PrivilegedActionException {
+      crm.grant("wizard", "gandalf");
+      Cache<?, ?> crmCache = extractField(crm, "clusterRoleMap");
+      ClusterTopologyManager crmTM = TestingUtil.extractComponent(crmCache, ClusterTopologyManager.class);
+      crmTM.setRebalancingEnabled(false);
+      await(cpm.addRole(Role.newRole("rogue", true, AuthorizationPermission.LISTEN)));
+      Cache<?, ?> cpmCache = extractField(cpm, "clusterPermissionMap");
+      ClusterTopologyManager cpmTM = TestingUtil.extractComponent(cpmCache, ClusterTopologyManager.class);
+      cpmTM.setRebalancingEnabled(false);
+      EmbeddedCacheManager joiner = Security.doAs(ADMIN, (PrivilegedExceptionAction<EmbeddedCacheManager>) () ->
+            addClusterEnabledCacheManager(getGlobalConfigurationBuilder(), getConfigurationBuilder(), new TransportFlags().withFD(true))
+      );
+      ClusterRoleMapper joinerCrm = Security.doAs(ADMIN, (PrivilegedExceptionAction<ClusterRoleMapper>) () -> (ClusterRoleMapper) joiner.getCacheManagerConfiguration().security().authorization().principalRoleMapper());
+      Set<String> admin0 = crm.list("gandalf");
+      assertTrue(admin0.contains("wizard"));
+      Set<String> admin2 = joinerCrm.list("gandalf");
+      assertFalse(admin2.contains("wizard"));
+      ClusterPermissionMapper joinerCpm = Security.doAs(ADMIN, (PrivilegedExceptionAction<ClusterPermissionMapper>) () -> (ClusterPermissionMapper) joiner.getCacheManagerConfiguration().security().authorization().rolePermissionMapper());
+      Role rogue0 = cpm.getRole("rogue");
+      assertNotNull(rogue0);
+      Role rogue2 = joinerCpm.getRole("rogue");
+      assertNull(rogue2);
+
+      crmTM.setRebalancingEnabled(true);
+      cpmTM.setRebalancingEnabled(true);
```

Comment:
probably should be in a try-finally block

Reply:
Done